msteams-platform/bots/how-to/bots-filesv4.md

```diff
@@ -0,0 +1,292 @@
+  * `personal`
+  * `channel`
+  * `groupchat`
+
+* Using the Teams bot APIs. These only support files in `personal` context.
+
+## Using the Graph APIs
+
+Post messages with card attachments that refer to existing SharePoint files, using the Graph APIs for [OneDrive and SharePoint](/onedrive/developer/rest-api/). To use the Graph APIs, obtain access to either of the following through the standard OAuth 2.0 authorization flow:
+* A user's OneDrive folder for `personal` and `groupchat` files.
+* The files in a team's channel for `channel` files. This method works in all Teams scopes.
+
+## Using the Teams bot APIs
+
+> [!NOTE]
+> Teams bot APIs work only in the `personal` context. They do not work in the `channel` or `groupchat` context.
+
+Using Teams APIs, the bot can directly send and receive files with users in the `personal` context, also known as personal chats. Implement features, such as expense reporting, image recognition, file archival, and e-signatures involving the editing of file content. Files shared in Teams typically appear as cards and allow rich in-app viewing.
+
+The following sections describe how to send file content as a direct user interaction, like sending a message. This API is provided as part of the Teams bot platform.
+
+### Configuring the bot to support files
+
+To send and receive files in the bot, set the `supportsFiles` property in the manifest to `true`. This property is described in the [bots](~/resources/schema/manifest-schema.md#bots) section of the Manifest reference.
+
+The definition looks like this: `"supportsFiles": true`. If the bot does not enable `supportsFiles`, the features listed in this section do not work.
+
+### Receiving files in personal chat
+
+When a user sends a file to the bot, the file is first uploaded to the user's OneDrive for Business storage. The bot then receives a message activity notifying the user about the user upload. The activity contains file metadata, such as its name and the content URL. The user can directly read from this URL to fetch its binary content.
+
+#### Message activity with file attachment example
+
+```json
+{
+  "attachments": [{
+    "contentType": "application/vnd.microsoft.teams.file.download.info",
+    "contentUrl": "https://contoso.sharepoint.com/personal/johnadams_contoso_com/Documents/Applications/file_example.txt",
+    "name": "file_example.txt",
+    "content": {
+      "downloadUrl" : "https://download.link",
+      "uniqueId": "1150D938-8870-4044-9F2C-5BBDEBA70C9D",
+      "fileType": "txt",
+      "etag": "123"
+    }
+  }]
+}
+```
+
+The following table describes the content properties of the attachment:
+
+| Property | Purpose |
+| --- | --- |
+| `downloadUrl` | OneDrive URL for fetching the content of the file. The user can issue an `HTTP GET` directly from this URL. |
+| `uniqueId` | Unique file ID. This is the OneDrive drive item ID, in case the user sends a file to the bot. |
+| `fileType` | File type, such as .pdf or .docx. |
```

Comment:
```Suggestion
| Property | Purpose |
| --- | --- |
| `downloadUrl` | OneDrive URL for fetching the content of the file. The user can issue an `HTTP GET` directly from this URL. |
| `uniqueId` | Unique file ID. This is the OneDrive drive item ID, in case the user sends a file to the bot. |
| `fileType` | Type of file, such as .pdf or .docx. |
```